src/site/components/up_to_top_button.html

```diff
@@ -0,0 +1,96 @@
+                        footer = document.querySelector('.footer-inner')
+                        // NOTE: Guarded Return so button does not show at all with errors 
+                        if(!threeFourthsContainer || !upToTopButton || !footer) return 'NODE_MISSING_CLASS' 
+                    }
+
+                    function scrollListener() {
+                        // Responsible for toggling animation classes 
+                        var scrollTop = (window.pageYOffset !== undefined) ? window.pageYOffset : (document.documentElement || document.body.parentNode || document.body).scrollTop;
+                        var buttonTransitionIn = 'vsa-top-button-transition-in'
+                        var buttonTransitionOut = 'vsa-top-button-transition-out'
+                        var buttonContainerRelative = 'vsa-top-button-container-relative'
+                        var buttonTransitionReset = 'vsa-top-button-transition-reset'
+                        
+                        
+                        if(scrollTop > 600 && !doesElHaveClass(upToTopButton, buttonTransitionIn)) {
+                            upToTopButton.classList.add(buttonTransitionIn)
+                            upToTopButton.classList.remove(buttonTransitionOut)
+                        } else if (scrollTop < 600 && doesElHaveClass(upToTopButton, buttonTransitionIn)) {
+                            upToTopButton.classList.add(buttonTransitionOut)
+                            upToTopButton.classList.remove(buttonTransitionIn)
+                        }
+
+                        if(isScrolledIntoView(footer)) {
+                            container.classList.add(buttonContainerRelative)
+                            threeFourthsContainer.classList.add(buttonContainerRelative)
+                            upToTopButton.classList.add(buttonTransitionReset)
+                        }
+                        else if(doesElHaveClass(container, buttonContainerRelative)) {
+                            container.classList.remove(buttonContainerRelative)
+                            threeFourthsContainer.classList.remove(buttonContainerRelative)
+                            upToTopButton.classList.remove(buttonTransitionReset)
+                        }
+
+                        function doesElHaveClass(el, className) {
+                            var hasClass = el.classList.value.indexOf(className) > -1
+                            return hasClass
+                        }
+
+                        function isScrolledIntoView(el) {
+                            var rect = el.getBoundingClientRect();
+                            var elemTop = rect.top;
+
+                            // Only partially || completely visible elements return true
+                            var isVisible = (elemTop >= 0) && (elemTop <= window.innerHeight);
+                            return isVisible;
+                        }
+                    }
```

Comment:
This is actually quite a bit more complex than I was expecting. Can you walk me through how this works/the various conditions? It looks like we are managing some state so that we can apply animations/transitions and we are also moving the button's position on screen based on screen size and relation to the footer?

I'm wondering if this is complex enough to warrant being moved into a JS bundle, so that we can take advantage of ES6 and polyfills.

Reply:
Oh interesting - per your screenshot, it should be of fixed position on screen but shouldn't overlap with the footer?

Reply:
Hey @ncksllvn !
Yeah Ryan and crew wanted the behavior mirrored closely to what button does here. 
https://www.vic.gov.au/digital-guides

As you scroll, you'll see a similar looking button appear, it's animations, and transfer from being fixed to relative just above the footer when a user scrolls past the footer.

Reply:
Haha okay, gotcha...that is a pretty slick button

Reply:
Yeah, I do enjoy me so animate buttons!